xarray/util/generate_aggregations.py

```diff
@@ -344,6 +399,40 @@ def generate_example(self, method):
 
         >>> {calculation}(){extra_examples}"""
 
+    def generate_cum_example(self, method):
```

Comment:
I agree it should be simpler, but I think how cumsum/cumprod work with multiple dimensions can be a bit confusing, so it is nice to have a multidimensional example.
maybe something like this?

```
>>> [[1.0, 1.0, 1.0],
...  [2.0, 2.0, 2.0],
...  [3.0, 3.0, np.nan]
```

Would add more but it would also be much easier to visualize by doing 

```
>>> da.cumprod('x')
>>> da.cumprod(['x', 'y'])
```

Reply:
> cumsum/cumprod work with multiple dimensions

This applies to normal reductions too. So maybe we can update all the examples in a future PR.